server/ai_session.go

```diff
@@ -222,7 +233,17 @@ func (sel *AISessionSelector) Select(ctx context.Context) *AISession {
 	shouldRefreshSelector := func() bool {
 		// Refresh if the # of sessions across warm and cold pools falls below the smaller of the maxRefreshSessionsThreshold and
 		// 1/2 the total # of orchs that can be queried during discovery
-		discoveryPoolSize := sel.node.OrchestratorPool.Size()
+		discoveryPoolSize := int(math.Min(float64(sel.node.OrchestratorPool.Size()), float64(sel.initialPoolSize)))
+
+		if (sel.warmPool.Size() + sel.coldPool.Size()) == 0 {
+			// release all orchestrators from suspension and try refresh
+			// if there are no orchestrators in the pools
+			clog.Infof(ctx, "refreshing sessions, no orchestrators in pools")
+			for i := 0; i < sel.penalty; i++ {
+				sel.suspender.signalRefresh()
```

Comment:
> release all orchestrators

shouldn't we then just remove them from the `suspender.list()` rather than calling `signalRefresh()`? My understanding is that if `penalty = 3`, then we would need to call `signalRefresh()` 3 times in order to "release all orchestrators from suspension".